storage-dynamodb/src/main/java/io/delta/storage/DynamoDBLogStore.java

```diff
@@ -16,10 +16,242 @@
+        if(!overwrite) {
+            Map<String, ExpectedAttributeValue> expected = new ConcurrentHashMap<>();
+            expected.put("fileName", new ExpectedAttributeValue(false));
+            pr.withExpected(expected);
+        }
+        return pr;
+    }
+
+    private void tryEnsureTableExists(Configuration hadoopConf) throws IOException {
+        int retries = 0;
+        boolean created = false;
+        while(retries < 20) {
+            String status = "CREATING";
+            try {
+                // https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/dynamodbv2/model/TableDescription.html#getTableStatus--
+                DescribeTableResult result = client.describeTable(tableName);
+                TableDescription descr = result.getTable();
+                status = descr.getTableStatus();
+            } catch (ResourceNotFoundException e) {
+                long rcu = Long.parseLong(getParam(hadoopConf, "provisionedThroughput.rcu", "5"));
+                long wcu = Long.parseLong(getParam(hadoopConf, "provisionedThroughput.wcu", "5"));
+
+                LOG.info(String.format(
+                    "DynamoDB table `%s` in region `%s` does not exist."
+                    + "Creating it now with provisioned throughput of %s RCUs and %s WCUs.",
+                    tableName, regionName, rcu, wcu));
+                try {
+                    client.createTable(
+                        // attributeDefinitions
+                        java.util.Arrays.asList(
+                            new AttributeDefinition("tablePath", ScalarAttributeType.S),
+                            new AttributeDefinition("fileName", ScalarAttributeType.S)
+                        ),
+                        tableName,
+                        // keySchema
+                        java.util.Arrays.asList(
+                            new KeySchemaElement("tablePath", KeyType.HASH),
+                            new KeySchemaElement("fileName", KeyType.RANGE)
+                        ),
+                        new ProvisionedThroughput(rcu, wcu)
+                    );
+                    created = true;
+                } catch(ResourceInUseException e3) {
+                    // race condition - table just created by concurrent process
+                }
+            }
+            if(status.equals("ACTIVE")) {
+                if(created) {
+                    LOG.info(String.format("Successfully created DynamoDB table `%s`", tableName));
+                } else {
+                    LOG.info(String.format("Table `%s` already exists", tableName));
+                }
+                break;
+            } else if(status.equals("CREATING")) {
+                retries += 1;
+                LOG.info(String.format("Waiting for `%s` table creation", tableName));
+                try {
+                    Thread.sleep(1000);
+                } catch(InterruptedException interrupted) {}
```

Comment:
why ignore the interrupt?

Reply:
Good point - probably because I didn't know how to handle it properly :) 
We have two options here:
* add `throws InterruptedException` to DynamoDBLogStore constructor signature
* catch this exception and wrap in IOException

@scottsand-db what do you think?

Reply:
```
        } catch (java.lang.InterruptedException e) {
            throw new InterruptedIOException(e.getMessage());
        }
```

just like here: https://github.com/delta-io/delta/pull/995/files#diff-fb8d246fe69244956c7da6b0468aa188c5c8964d26ea6bf1ef1811cf1c31b598R307-R309